week3/root/SpeechPractice/SpeechPractice/SpeechRecognizer.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  SpeechRecognizer.swift
+//  SpeechPractice
+//
+//  Created by JiJooMaeng on 6/21/25.
+//
+
+
+import SwiftUI
+import Speech
+import AVFoundation
+
+class SpeechRecognizer: ObservableObject {
+    private let speechRecognizer = SFSpeechRecognizer(locale: Locale(identifier: "ko-KR"))  //en-US, ko-KR
+    private var recognitionRequest: SFSpeechAudioBufferRecognitionRequest?
+    private var recognitionTask: SFSpeechRecognitionTask?
+    private let audioEngine = AVAudioEngine()
+    
+    @Published var transcribedText: String = ""
+    @Published var isRecording: Bool = false
+
+    func startRecording() {
+        SFSpeechRecognizer.requestAuthorization { authStatus in
+            DispatchQueue.main.async {
+                if authStatus == .authorized {
+                    self.record()
+                } else {
+                    self.transcribedText = "음성 인식 권한이 없습니다."
+                }
+            }
+        }
+    }
+
+    private func record() {
+        recognitionTask?.cancel()
+        recognitionTask = nil
+
+        let audioSession = AVAudioSession.sharedInstance()
+        do {
+            try audioSession.setCategory(.record, mode: .measurement, options: .duckOthers)
+            try audioSession.setActive(true, options: .notifyOthersOnDeactivation)
+        } catch {
+            transcribedText = "오디오 세션 설정 실패: \(error.localizedDescription)"
+            return
+        }
+
+        recognitionRequest = SFSpeechAudioBufferRecognitionRequest()
+
+        guard let inputNode = audioEngine.inputNode as Optional else {
+            transcribedText = "입력 노드 오류"
+            return
+        }
```

Comment:
```suggestion
        let inputNode = audioEngine.inputNode
```

이 부분 Optional로 캐스팅하는 이유가 있을까요?

Reply:
먼가 오류에 대한 예외처리를 하려고 했던거 같은데, 다시보니 억지로 Optional로 캐스팅해서 guard let 바이딩 시키고 있는거 같네요. 바꿔도 좋을 것 같습니다!